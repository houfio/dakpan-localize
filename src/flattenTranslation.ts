import { Translations } from './types';

export const flattenTranslation = (translation: Translations) => {
  let output = {};

  const step = (t: Translations, previous = '') => {
    Object.keys(t).forEach((key) => {
      const value = t[key];
      const path = previous ? `${previous}.${key}` : key;

      if (!(value instanceof Array)) {
        return step(value, path);
      }

      output = {
        ...output,
        [path]: value
      };
    });
  };

  step(translation);

  return output;
};
