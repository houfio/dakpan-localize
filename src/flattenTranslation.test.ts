import { flattenTranslation } from './flattenTranslation';

it('should flatten the object', () => {
  expect(flattenTranslation({
    message: ['hello'],
    form: {
      email: ['Email address'],
      password: ['Password'],
      other: {
        test: ['awdwadawd']
      }
    }
  })).toMatchSnapshot();
});
