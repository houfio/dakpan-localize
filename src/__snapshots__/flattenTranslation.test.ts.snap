// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should flatten the object 1`] = `
Object {
  "form.email": Array [
    "Email address",
  ],
  "form.other.test": Array [
    "awdwadawd",
  ],
  "form.password": Array [
    "Password",
  ],
  "message": Array [
    "hello",
  ],
}
`;
